### ISSUE TITLE
Allure modules: allure-model, allure-report-data and etc. should use slf4j-api dependency rather than slf4j-log4j12

### ISSUE THREAD

**Issue by valfirst** [allure-framework/allure-core#564]
Now allure modules have the following dependency:
```xml
<dependency>
        <groupId>org.slf4j</groupId>
        <artifactId>slf4j-log4j12</artifactId>
</dependency>
```
It's recommended to use slf4j-api which allows the end users to plug any desired logging framework (http://www.slf4j.org/manual.html).

**Comment by vania-pooh** [comment]
@valfirst what is the use case for that? Do you really have an intention to use some Allure modules in your projects?

**Comment by valfirst** [comment]
@vania-pooh, I'm implementing Allure adapter for JBeahve, during implementation I've found out that set of different allure modules depends on org.slf4j:slf4j-log4j12. It brought some inconvenience for me, since I was needed to exclude this module on my side, because I want to use log4j2 instead of log4j v1. According to the specification of slf4j it's recommended to use slf4j-api, so that users may choose logging framework on their side. That's my issue is a kind of recommendation for all Allure modules.

**Comment by smecsia** [comment]
@vania-pooh I agree with @valfirst. We discussed this topic much earlier and guys promised to replace this dependency with `sl4j-api`. It seems that they haven't done this yet.

**Comment by vania-pooh** [comment]
@smecsia, @valfirst  I would say a pull request from your side will definitely accelerate this.

**Comment by baev** [comment]
@valfirst could you please send us the pull request and fix it?

**Comment by baev** [comment]
fixed via #597 #609
